### PR TITLE
Gitactions to automatically update and push SPM BIDS-Prov examples

### PR DESCRIPTION
**Pull request by cyril-data**
PR to add github actions to transform all SPM task-fMRI result examples available in https://github.com/incf-nidash/nidmresults-examples in jsonld with spm_parser. The results are pushed in the directory *"results"*

To do so the principal important changes are for the files : 
- *".github/workflows/SPM_test_on_nidmexamples.yml"* : to be able for the github bot to execute git commands (`git add`, `git commit`, `git push`)
- *"bids_prov/tests/test_spm_parser.py"* and *"bids_prov/tests/samples_test/partial_conjunction_ref.jsonld"*: impose the initialization of random seed for **each** `batch.m` on dir *bids_prov/tests/samples_test/*. In consequence, IDs of file *"bids_prov/tests/samples_test/partial_conjunction_ref.jsonld"* have changed. 
- *"launch_multiple_spm.py"* : Add a context file in dir resuts 
- *"results"* : all the files computed by the github actions

### CODE REVIEW

**Comment by cmaumet** [comment]
Thanks @cyril-data! I've amended the comment and title of this PR to add links to the original dataset used to create examples. Could you try and solve the check that fails? And then I'll review the PR. Thank you!

**Comment by cyril-data** [comment]
Hi Camille, 

The gitaction work definitely on the fork but not on the bids-standard... 

With some test, I got this error : 

```
 remote: Permission to Inria-Empenn/BEP028_BIDSprov.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/Inria-Empenn/BEP028_BIDSprov/': The requested URL returned error: 403
Error: Process completed with exit code 128.
```
Could you cheked here : https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#enabling-workflows-for-private-repository-forks

to see in section (Enabling workflows for forks ...) if access are allowed for fork repo ? 

Thanks a lot

**Comment by thomasbtnfr** [comment]
The action is only triggered by push now

File: results/context_spm_2023_01_05_14h41m40s.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Date : 2023_01_05_14h41m40s
```
Comment (cmaumet) [inline results/context_spm_2023_01_05_14h41m40s.txt:1]: Is this file useful to have in git?

**Comment by cmaumet** [comment]
Thanks @thomasbtnfr @cyril-data! This looks good to me, let's merge :)

**Pull request merged**